Generate Java source file for Android from the list of logs.

plietar opened the pull request:
This file will be included in conscrypt for Android

plietar commented:
Generated output :
https://android-review.googlesource.com/#/c/171932/6/src/main/java/org/conscrypt/ct/KnownLogs.java

DO NOT MERGE YET
I may need to change the format based on reviews on the Android side.

eranmes commented on python/utilities/log_list/print_log_list.py:
```diff
@@ -131,6 +134,75 @@ def generate_code_for_chromium(json_log_list, output_file):
     f.write("\nconst size_t kNumKnownCTLogs = %d;\n" % len(logs))
     write_cpp_footer(f, include_guard)
 
+def write_android_header(f):
```
Break this into a separate module (within this directory).

eranmes replied:
While you're at it, please look into breaking the C++ code generation to a module of its own.

plietar commented:
I've changed the output format to be more generic rather than just file conscrypt.
The package and class names still default to conscrypt, but can be overridden with a flag.
The copyright notice is still the one from AOSP.

Also split out the code generation is separate modules

eranmes commented on python/utilities/log_list/java_generator.py:
```diff
@@ -0,0 +1,73 @@
+    f.write(
+        "public final class %(cls)s {\n"
+        "    public static final int LOG_COUNT = %(count)d;\n"
+        "    public static final String[] LOG_DESCRIPTIONS = new String[] {\n"
```
Seems like some lines here are too long, please run it through pylint.

eranmes commented on python/utilities/log_list/java_generator.py:
```diff
@@ -0,0 +1,73 @@
+    signed = (u - 256 if u > 127 else u for u in unsigned)
+    return ("        // " + description + "\n"
+            "        new byte[] {\n"
+            + textwrap.fill(", ".join("%d" % s for s in signed), width=85,
```
Could you assign the result from this call to a variable you'll include in the string generated here, so the code is slightly more readable?

eranmes commented on python/utilities/log_list/java_generator.py:
```diff
@@ -0,0 +1,73 @@
+        "%(keys)s"
+        "    };\n"
+        "}\n" % {
+            "cls": cls,
```
nit: Prefer a long name than a shorthand.

eranmes commented:
A few minor comments, LGTM otherwise - please rebase and let me know so I'll merge and commit.
